skyVegasTests/gamePlayGameClient.spec.ts

```diff
@@ -1,35 +1,261 @@
-import { test, type Page } from '@playwright/test';
+import { test, type Page, expect } from '@playwright/test';
 import {
     launchGame,
     login,
-    validateConsoleMessages,
     startEventListener,
-    makeSpin,
+    firstSpin,
     readGames,
+    spin,
+    getValueFromConsoleMessages,
+    currencyStringToNumber,
 } from './utilities';
 import { ExpectedMessage } from '../types/expectedMessage';
 
 let page: Page;
 readGames().forEach((game) => {
-    const consoleMessages: string[] = [];
-
     test.describe(`Testing with text: ${game}`, () => {
-        test.beforeAll(async ({ browser }) => {
+        test.beforeEach(async ({ browser }) => {
+            test.setTimeout(300000);
             page = await browser.newPage();
             await login(page);
         });
 
-        test.afterAll(async () => {
+        test.afterEach(async () => {
             await page.close();
         });
 
-        test('gamePlay', async () => {
+        test('gamePlayForLoss', async () => {
+            const consoleMessages: string[] = [];
+            let numberOfSpins = 0;
+            startEventListener(page, consoleMessages);
+
+            await launchGame(page, game, consoleMessages);
+
+            const startBalanceConsole = await getValueFromConsoleMessages(
+                consoleMessages,
+                ExpectedMessage.BALANCE_UPDATE
+            );
+
+            const startBallanceScreen = currencyStringToNumber(
+                await page
+                    .frameLocator('#root iframe')
+                    .locator('em.value.main-wallet')
+                    .textContent()
+            );
+
+            const stakeAmountConsole = await getValueFromConsoleMessages(
+                consoleMessages,
+                ExpectedMessage.STAKE_UPDATE
+            );
+
+            const stakeAmountScreen = currencyStringToNumber(
+                await page
+                    .frameLocator('#root iframe')
+                    .locator('em.value.stake-wallet')
+                    .textContent()
+            );
+
+            await firstSpin(page, consoleMessages);
+            numberOfSpins++;
+
+            let totalWinAmountConsole = 0;
```

Comment:
the next section has some back to back code repetition, it could probably be refactored, probably around that while condition to cut down the need for the section of it beforehand

Reply:
Will make this change in the PR for GTX-441. We had to pull that work into this one as there were overlaps with helper functions that were needed.